src/java/org/apache/cassandra/gms/FailureDetector.java

```diff
@@ -93,16 +94,14 @@ public FailureDetector()
 
     private static long getInitialValue()
     {
-        String newvalue = System.getProperty("cassandra.fd_initial_value_ms");
-        if (newvalue == null)
-        {
+        Long newValue = FD_INITIAL_VALUE_MS.getLong();
+        if (newValue == null)
             return Gossiper.intervalInMillis * 2;
-        }
-        else
-        {
-            logger.info("Overriding FD INITIAL_VALUE to {}ms", newvalue);
-            return Integer.parseInt(newvalue);
-        }
+
+        if (newValue != Gossiper.intervalInMillis * 2)
```

Comment:
As I mentioned it here:
https://github.com/apache/cassandra/pull/2046#discussion_r1170095535

The `isDefault` will not work because the `Gossiper.intervalInMillis * 2` is calculated at runtime, so there is no need to compare it with the `defaultVal`, which is always null in such cases. So we either move the default value to `FD_INITIAL_VALUE_MS` (removing the dependency on `Gossiper.intervalInMillis`) or leave everything 'as is', as there are only a few cases with such a usage pattern. I think the latter is the best option for us as it has no impact on the API.